src/buildx.ts

```diff
@@ -106,6 +107,43 @@ export async function inspect(name: string): Promise<Builder> {
     });
 }
 
+export async function build(inputBuildRef: string, dockerConfigHome: string): Promise<string> {
+  let [repo, ref] = inputBuildRef.split('#');
+  if (ref.length == 0) {
+    ref = 'master';
+  }
+
+  const sha = await git.getRemoteSha(repo, ref);
+  core.debug(`Remote ref ${sha} found`);
+
+  let toolPath: string;
+  toolPath = tc.find('buildx', sha);
+  if (!toolPath) {
+    const outFolder = path.join(context.tmpDir(), 'out').split(path.sep).join(path.posix.sep);
+    toolPath = await exec
+      .getExecOutput(
+        'docker',
+        ['buildx', 'build', '--target', 'binaries', '--output', `type=local,dest=${outFolder}`, inputBuildRef],
+        {
+          ignoreReturnCode: true
+        }
+      )
+      .then(res => {
+        if (res.stderr.length > 0 && res.exitCode != 0) {
+          core.warning(res.stderr.trim());
+        }
+        return tc.cacheFile(
+          `${outFolder}/buildx`,
+          context.osPlat == 'win32' ? 'docker-buildx.exe' : 'docker-buildx',
+          'buildx',
+          sha
+        );
```

Comment:
Binary will be cached. Atm remote ref sha is used as cache key. `${repo}-${sha}` would be safer imo.